internal/pkg/warmup/warmup.go

```diff
@@ -112,3 +118,13 @@ func (w Warmup) GrpcWarmupWorker(wg *sync.WaitGroup, requests <-chan grpc.Reques
 	}
 	wg.Done()
 }
+
+// checks if ramp up is enabled, sleep not required for concurrency 1, sleep may not be required for last few ramp ups (when unbalanced interval)
+func waitForRampUp(rampUpIntervalSeconds int, currentConcurrency int, endTime time.Time) bool {
```

Comment:
i think this is a bit confusing from a user point of view and prone to wrong configurations

i suggest a different approach where instead of defining an interval for increasing the concurrency we define the time it should take for mittens to be at max concurrency

for example, setting this to 60s would mean mittens would take 60s to reach max concurrency. 
this would also make the code simpler to implement as we can simply validate that this value is less than the warmup duration and then the sleep would be max concurrency / this value

Reply:
I agree, this is indeed complex and leads to weird situations which needed handling.

I will close this pr and create a new one. Thank for the inputs @worldtiki and @nikos912000